frontend/src/bundles/candidate/components/candidate-modal/candidate-modal.tsx

```diff
@@ -0,0 +1,305 @@
+    const { messageTemplates } = useAppSelector(({ candidate }) => ({
+        messageTemplates: candidate.messageTemplates,
+    }));
+
+    const { control, errors, handleSubmit, setValue } =
+        useAppForm<ContactCandidateDto>({
+            defaultValues: DEFAULT_CONTACT_CANDIDATE_MODAL,
+            validationSchema: ContactCandidateValidationSchema,
+        });
+
+    const { fields, append, remove } = useFieldArray({
+        name: 'links',
+        control,
+    });
+
+    const addLink = useCallback((): void => {
+        append({
+            value: '',
+        });
+    }, [append]);
+
+    const removeLink = useCallback(
+        (index: number) => (): void => {
+            remove(index);
+        },
+        [remove],
+    );
+
+    const applyTemplate = useCallback(
+        (message: string) => (): void => {
+            setValue('message', message);
+        },
+        [setValue],
+    );
+
+    const onSubmit: SubmitHandler<ContactCandidateDto> = useCallback(
+        (data: ContactCandidateDto): void => {
+            if (data.templateName) {
+                const { templateName: name, message } = data;
+                void dispatch(
+                    candidateActions.addMessageTemplate({ name, message }),
+                );
+            }
+            // eslint-disable-next-line no-console
+            console.log(data); // remove later with real logic
+        },
+        [dispatch],
+    );
+
+    const handleCheckboxOnChange = useCallback(
+        (props: ControllerRenderProps<ContactCandidateDto, 'isSaveTemplate'>) =>
+            (event: React.ChangeEvent<HTMLInputElement>): void => {
+                props.onChange(event.target.checked);
+                setIsSaveTemplateChecked(event.target.checked);
+            },
+        [],
+    );
+
+    const renderCheckbox = useCallback(
+        ({
+            field: props,
+        }: {
+            field: ControllerRenderProps<ContactCandidateDto, 'isSaveTemplate'>;
+        }): React.ReactElement => {
+            return (
+                <Checkbox
+                    name={props.name}
+                    checked={props.value}
+                    className={styles.checkbox}
+                    onChange={handleCheckboxOnChange(props)}
+                    label="Save as a template"
+                />
+            );
+        },
+        [handleCheckboxOnChange],
+    );
+
+    const handleFormSubmit = useCallback(
+        (event_: React.BaseSyntheticEvent): void => {
+            void handleSubmit(onSubmit)(event_);
+        },
+        [handleSubmit, onSubmit],
+    );
+
+    const { control: templateControl, errors: templateErrors } = useAppForm({
+        defaultValues: { templates: messageTemplates },
+    });
+
+    return (
+        <Modal
+            isOpen={isOpen}
+            onClose={onClose}
+            headerLabel="Contact candidate"
+        >
+            <form className={styles.form} onSubmit={handleFormSubmit}>
```

Comment:
FormControl

Reply:
@nikita-remeslov why do we have to use FormControl? Our sign-in/sign-up pages use default <form> tag